core/tools/blueos_startup_update/blueos_startup_update.py

```diff
@@ -404,6 +404,26 @@ def fix_ssh_ownership() -> bool:
     return False
 
 
+def fix_wpa_service() -> bool:
+    """
+    Adds -i wlan0 and -c /etc/wpa_supplicant/wpa_supplicant.conf to the wpa_supplicant service
+    This is needed to make the service actually consume the .conf file with update_config=1
+    """
+    logger.info("checking wpa_supplicant service...")
+    file_path = "/lib/systemd/system/wpa_supplicant.service"
+    original_file = load_file(file_path)
+    # extract execstart line
+    execstart_line = next((line for line in original_file.splitlines() if line.startswith("ExecStart=")), None)
+    if execstart_line and "-i " in execstart_line and "-c " in execstart_line:
+        # the settings are there. not our job to check if they are the ones we want
+        return False
+    original_file = original_file.replace(
+        execstart_line, f"{execstart_line} -i wlan0 -c /etc/wpa_supplicant/wpa_supplicant.conf"
```

Comment:
this can end up duplicating -i or -c if only one of them is in the original line